### PR TITLE
test: Catalog acceptances tests refactor

### PR DESCRIPTION
**Pull request by azrod**
### Description of your changes

Refactor catalog acceptances tests
Due #489 
Block #495

### CODE REVIEW

**Comment by valentin-dubreuil** [comment]
## :bangbang: Acceptance Test failed 

=== RUN   TestCatalogAccACLResource
    catalog_acl_resource_test.go:122: Step 1/4 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        Provider "registry.terraform.io/hashicorp/cloudavenue" requires explicit
        configuration. Add a provider block to the root module and configure the
        provider's required arguments as described in the provider documentation.
        
        
        Error: Unable to Create Cloud Avenue API Client
        
          with provider["registry.terraform.io/hashicorp/cloudavenue"],
          on <empty> line 0:
          (source code not available)
        
        An unexpected error occurred when creating the Cloud Avenue API client. If
        the error is not clear, please contact the provider developers.
        
        Cloud Avenue Client Error: authentication error : Post
        "https://console1.cloudavenue.orange-business.com/cloudapi/1.0.0/sessions":
        tls: failed to verify certificate: x509: certificate signed by unknown
        authority
--- FAIL: TestCatalogAccACLResource (0.79s)
FAIL
FAIL	github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc	0.799s
FAIL

**Comment by valentin-dubreuil** [comment]
## :bangbang: Acceptance Test failed 

=== RUN   TestCatalogAccACLResource
    catalog_acl_resource_test.go:122: Step 1/4 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        Provider "registry.terraform.io/hashicorp/cloudavenue" requires explicit
        configuration. Add a provider block to the root module and configure the
        provider's required arguments as described in the provider documentation.
        
        
        Error: Unable to Create Cloud Avenue API Client
        
          with provider["registry.terraform.io/hashicorp/cloudavenue"],
          on <empty> line 0:
          (source code not available)
        
        An unexpected error occurred when creating the Cloud Avenue API client. If
        the error is not clear, please contact the provider developers.
        
        Cloud Avenue Client Error: authentication error : Post
        "https://console1.cloudavenue.orange-business.com/cloudapi/1.0.0/sessions":
        tls: failed to verify certificate: x509: certificate signed by unknown
        authority
--- FAIL: TestCatalogAccACLResource (0.71s)
FAIL
FAIL	github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc	0.714s
FAIL

**Comment by valentin-dubreuil** [comment]
## :bangbang: Acceptance Test failed 

=== RUN   TestCatalogAccACLResource
    catalog_acl_resource_test.go:122: Step 1/4 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        Provider "registry.terraform.io/hashicorp/cloudavenue" requires explicit
        configuration. Add a provider block to the root module and configure the
        provider's required arguments as described in the provider documentation.
        
        
        Error: Unable to Create Cloud Avenue API Client
        
          with provider["registry.terraform.io/hashicorp/cloudavenue"],
          on <empty> line 0:
          (source code not available)
        
        An unexpected error occurred when creating the Cloud Avenue API client. If
        the error is not clear, please contact the provider developers.
        
        Cloud Avenue Client Error: authentication error : Post
        "https://console1.cloudavenue.orange-business.com/cloudapi/1.0.0/sessions":
        tls: failed to verify certificate: x509: certificate signed by unknown
        authority
--- FAIL: TestCatalogAccACLResource (0.90s)
FAIL
FAIL	github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc	0.909s
FAIL

**Comment by azrod** [comment]
--- PASS: TestCatalogAccACLDataSource (33.17s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  33.776s

**Comment by azrod** [comment]
=== RUN   TestAccCatalogResource
--- PASS: TestAccCatalogResource (35.61s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  36.141s

File: internal/testsacc/acctest_resources_test.go
Comment (dmicheneau) [inline internal/testsacc/acctest_resources_test.go:1]: I see this GetResource will be generic on next PR

**Review by dmicheneau — APPROVED** [review]

**Pull request merged**